frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -1538,4 +1661,367 @@ function ReaderRolling:onToggleHideNonlinear()
+                        return
+                    end
+                    -- Otherwise, conditions not yet met
+                else -- self.RENDERING_STATE.RELOADING_DOCUMENT
+                    if not do_reload then
+                        -- Stuff happened: transition back to previous state
+                        self._stepRerenderingAutomation(self.RENDERING_STATE.FULL_RENDERING_READY)
+                    else
+                        -- reload icon shown, ready to reload
+                        if self._current_rerendering_pid and not self:_waitOrKillCurrentRerenderingSubprocess(false) then
+                            -- Rendering subproces still alive, waiting for our signal to save its cache
+                            -- We need to let the subprocess save the cache, and only then reloadDocument.
+                            -- We need to block the UI so no other setting get changed, and the cache
+                            -- is sane to reload from.
+                            -- Let subprocess know it can save cache
+                            shared_state[2] = 1
+                            -- And wait for it to end (successfully or not)
+                            repeat
+                                ffiutil.usleep(250000)
+                            until self:_waitOrKillCurrentRerenderingSubprocess(false)
+                        end
+                        -- Otherwise, no background rerendering needed, or the subprocess died: go on with the reload.
+                        -- We're done with background stuff and icon animations: reallow standby
+                        self.rendering_state = self.RENDERING_STATE.DO_RELOAD_DOCUMENT
+                        self.ui:reloadDocument(nil, true) -- seamless reload (no infomsg, no flash)
+                    end
+                    return
+                end
+            end
+        end
+        if self.rendering_state ~= prev_state then
+            logger.dbg("_stepRerenderingAutomation", prev_state, ">", self.rendering_state)
+            -- Have ReaderView redraw and refresh ReaderFlipping and our state icon, avoiding flashes
+            UIManager:setDirty(self.view.dialog, "ui", self.view.flipping.dimen)
+        end
+        if reschedule then
+            UIManager:scheduleIn(1, self._stepRerenderingAutomation)
+        end
+    end
+end
+
+function ReaderRolling:tearDownRerenderingAutomation()
+    if self._stepRerenderingAutomation then
+        UIManager:unschedule(self._stepRerenderingAutomation)
+        self._stepRerenderingAutomation = nil
+        UIManager:allowStandby()
+    end
+    if self._watchInputEvent then
+        UIManager.event_hook:unregister("InputEvent", self._watchInputEvent)
+        self._watchInputEvent = nil
+    end
+    -- Be sure we don't let any zombie
+    self:_waitOrKillCurrentRerenderingSubprocess(true, true)
+    Device:enableCPUCores(1)
+end
+
+function ReaderRolling:_rerenderInBackground()
+    Device:enableCPUCores(2)
```

Comment:
As I see KOReader is mostly single core by now. But hard coding `2` here might be not so clever for the future.
Maybe you could add something like `Device:getEnabledCPUCores()` and use `Device:enabledCPUCores(Device:getEnabledCPUCores() + 1)` and `Device:enabledCPUCores(Device:getEnabledCPUCores() - 1)` a few lines above?
    
Or even better a method like `Device:adjustEnabledCPUCores(+1)` and `Device:adjustEnabledCPUCores(-1)`?

Reply:
Could probably make sense, or a bit like allow/preventStandby, incrementing/decrementing a counter so one "allow" doesn't cancel another "prevent" done somewhere else.
So, something like Device:gimmeOneMoreCore().
But not for me (without a multicore device) to try it, and not for this PR.

Reply:
While it sounds sensible in theory, given that Lua is single-threaded by nature, this won't ever make sense in practice ;).